src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -136,10 +138,34 @@ SELECT CASE WHEN (count(c) > 0) THEN true ELSE false END
     @EntityGraph(type = LOAD, attributePaths = { "exercises", "lectures", "lectures.lectureUnits", "learningGoals", "prerequisites" })
     Optional<Course> findWithEagerExercisesAndLecturesAndLectureUnitsAndLearningGoalsById(long courseId);
 
-    @Query("select distinct course from Course course left join fetch course.organizations organizations left join fetch course.prerequisites prerequisites where (course.startDate is null or course.startDate <= :now) and (course.endDate is null or course.endDate >= :now) and course.onlineCourse = false and course.registrationEnabled = true")
-    List<Course> findAllCurrentlyActiveNotOnlineAndRegistrationEnabledWithOrganizationsAndPrerequisites(@Param("now") ZonedDateTime now);
+    @Query("""
+                SELECT course
+                FROM Course course
+                LEFT JOIN FETCH course.organizations organizations
+                LEFT JOIN FETCH course.prerequisites prerequisites
```

Comment:
also for the other instances in this class.
```suggestion
                    LEFT JOIN FETCH course.organizations organizations
                    LEFT JOIN FETCH course.prerequisites prerequisites
```